tests/unit_tests/test_tokenizer.py

```diff
@@ -0,0 +1,410 @@
+                our_tokens,
+                reference_tokens,
+                f"Edge case tokens should match for text '{text[:50]}...' in {test_repo_id} ({tokenizer_type})",
+            )
+
+    @parametrize(
+        "test_repo_id",
+        [
+            "meta-llama/Meta-Llama-3.1-8B",
+            "deepseek-ai/DeepSeek-V3",
+            # "black-forest-labs/FLUX.1-dev", TODO: load the actual tokenizer
+            "Qwen/Qwen2-7B",
+        ],
+    )
+    def test_download_and_load_tokenizer_integration(self, test_repo_id):
+        """
+        Test downloading tokenizer files and loading them, comparing with official APIs.
+
+        This test:
+        1. Downloads tokenizer files using download_hf_tokenizer_files
+        2. Loads tokenizer using our load_tokenizer function
+        3. Compares behavior with official Tokenizer library
+        4. Compares with transformers AutoTokenizer (if available)
+        """
+        # Step 1: Download tokenizer files
+        try:
+            download_hf_tokenizer_files(
+                repo_id=test_repo_id,
+                local_dir=self.temp_dir,
+            )
+        except HTTPError as e:
+            if test_repo_id == "meta-llama/Meta-Llama-3.1-8B":
+                self.skipTest(
+                    f"Could not download tokenizer files for Meta-Llama-3.1-8B: {e}"
+                )
+            else:
+                raise e
+
+        # Step 2: Load tokenizer using our function
+        model_name = test_repo_id.split("/")[-1]
+        tokenizer_dir = "tokenizer" if model_name == "FLUX.1-dev" else "."
+        tokenizer_path = os.path.join(self.temp_dir, model_name, tokenizer_dir)
+        our_tokenizer = build_hf_tokenizer(tokenizer_path)
+
+        # Step 3: Load tokenizer using official Tokenizer library (if available)
+        official_tokenizer = None
+        try:
+            official_tokenizer = Tokenizer.from_pretrained(test_repo_id)
+        except Exception as e:
+            print(f"Warning: Could not load official tokenizer for {test_repo_id}: {e}")
+
+        # Step 4: Load tokenizer using transformers AutoTokenizer (if available)
+        transformers_tokenizer = None
+        try:
+            from transformers import AutoTokenizer
+
+            transformers_tokenizer = AutoTokenizer.from_pretrained(test_repo_id)
+        except Exception as e:
+            print(f"Warning: Could not load AutoTokenizer for {test_repo_id}: {e}")
+
+        # Step 5: Compare underlying tokenizer attributes (only if official tokenizer is available)
+        if official_tokenizer:
+            self._compare_tokenizers(
+                our_tokenizer.tokenizer, official_tokenizer, test_repo_id
+            )
+
+        # Step 6: Compare with transformers tokenizer if available
+        if transformers_tokenizer:
+            self._compare_tokenizers(
+                our_tokenizer, transformers_tokenizer, test_repo_id
+            )
+
+    def test_backward_comptability(self):
```

Comment:
validating that `TikTokenizer` (old implementation) encode and decodes the same as `HuggingFaceTokenizer` (new implementation). Will remove this test after we remove TikTokenizer